### PR TITLE
Bug/308 filter

### PR DESCRIPTION
**Pull request by timohuber**
- Do not allow empty AND or OR filters
- Make notifications of filter form removable
- Add delete handler to experiment filters
- Delete filter when experiment is deleted

### CODE REVIEW

File: pool/cqrs_command/experiment_command.ml [hunk 1]
```diff
@@ -145,16 +145,25 @@ end = struct
...
+  let handle ?(tags = Logs.Tag.empty) { experiment; session_count } =
     Logs.info ~src (fun m -> m "Handle command Delete" ~tags);
     match session_count > 0 with
     | true -> Error Pool_common.Message.ExperimentSessionCountNotZero
     | false ->
-      Ok [ Experiment.Destroyed experiment_id |> Pool_event.experiment ]
+      let filter_event =
+        experiment.Experiment.filter
+        |> CCOption.map_or ~default:[] (fun f ->
+             [ Filter.Deleted f |> Pool_event.filter ])
```
Comment (mabiede) [inline pool/cqrs_command/experiment_command.ml:160]: isn't there a constructor for `Filter.deleted` --> `CCFun.(Filter.deleted %> Pool_event.filter %> CCList.pure`

File: pool/cqrs_command/experiment_command.ml [hunk 2]
```diff
@@ -145,16 +145,25 @@ end = struct
...
+      in
+      Ok
+        ([ Experiment.Destroyed experiment.Experiment.id
```
Comment (mabiede) [inline pool/cqrs_command/experiment_command.ml:163]:
```suggestion
        Ok (Experiment.Destroyed experiment.Experiment.id |> Pool_event.experiment :: filter_event)
```

File: pool/routes/routes.ml [hunk 3]
```diff
@@ -379,8 +379,12 @@ module Admin = struct
         Handler.Admin.
           [ post
               "/create"
-              ~middlewares:[ Experiments.Access.create ]
+              ~middlewares:[ Experiments.Access.update ]
               Filter.create_for_experiment
+          ; post
+              "/delete"
+              ~middlewares:[ Experiments.Access.update ]
```
Comment (mabiede) [inline pool/routes/routes.ml:386]: should have "Access.delete"

File: pool/cqrs_command/experiment_command.ml (hunk 1 above)
Comment (timohuber) [reply pool/cqrs_command/experiment_command.ml:160]: I added the variants functions

**Review by mabiede — APPROVED** [review]

**Pull request merged**